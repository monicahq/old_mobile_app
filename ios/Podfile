# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'Monica' do
  pod 'AppCenter/Crashes', '~> 1.9.0'
  pod 'AppCenter/Analytics', '~> 1.9.0'
  pod 'AppCenterReactNativeShared', '~> 1.8.0'
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for Monica


  platform :ios, '9.0'
  target 'MonicaTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

target 'Monica-tvOS' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for Monica-tvOS

  target 'Monica-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

end
