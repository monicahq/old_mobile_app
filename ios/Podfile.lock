PODS:
  - AppCenter/Analytics (1.9.0):
    - AppCenter/Core
  - AppCenter/Core (1.9.0)
  - AppCenter/Crashes (1.9.0):
    - AppCenter/Core
  - AppCenterReactNativeShared (1.8.0):
    - AppCenter/Core (= 1.9.0)

DEPENDENCIES:
  - AppCenter/Analytics (~> 1.9.0)
  - AppCenter/Crashes (~> 1.9.0)
  - AppCenterReactNativeShared (~> 1.8.0)

SPEC REPOS:
  https://github.com/cocoapods/specs.git:
    - AppCenter
    - AppCenterReactNativeShared

SPEC CHECKSUMS:
  AppCenter: 3cf7b9387549e9703dab068793e404013e8c0cc1
  AppCenterReactNativeShared: 53eb73a85b3adaf0d5b6dd8ea276a56e2d9aed6c

PODFILE CHECKSUM: b0c19c98bd5484d6dd08724f52840eee715339ef

COCOAPODS: 1.5.3
