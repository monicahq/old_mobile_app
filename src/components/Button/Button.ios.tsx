import React, {PureComponent} from 'react';
import {ActivityIndicator, Text, TouchableOpacity} from 'react-native';
import {IButtonProps} from './Button.props';
import {styles} from './Button.styles';

export class Button extends PureComponent<IButtonProps> {
  public static defaultProps = {
    disabled: false,
    loading: false,
  };

  public getTitle() {
    const {title, children, loadingTitle, loading} = this.props;
    if (loading && loadingTitle) {
      return loadingTitle;
    }
    return title || children || '';
  }

  public render() {
    const {onPress, disabled, loading, id} = this.props;

    return (
      <TouchableOpacity
        testID={id}
        style={[styles.button, disabled && styles.buttonDisabled]}
        onPress={onPress}
        disabled={disabled}
        activeOpacity={0.4}
      >
        <Text style={styles.buttonText}>{this.getTitle()}</Text>
        {loading && (
          <ActivityIndicator color="#ffffff" style={styles.activityIndicator} />
        )}
      </TouchableOpacity>
    );
  }
}
