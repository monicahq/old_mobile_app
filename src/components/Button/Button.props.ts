export interface IButtonProps {
  id?: string;
  onPress: (...args) => any;
  children?: string;
  loading?: boolean;
  title?: string;
  disabled?: boolean;
  loadingTitle?: string;
}
