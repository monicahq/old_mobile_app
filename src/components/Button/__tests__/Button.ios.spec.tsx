import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Button} from '../Button.ios';

describe('Components', () => {
  describe('Button IOS', () => {
    const defaultOnPress = {
      onPress: () => {
        return;
      },
    };

    it('should renders correctly', () => {
      expect(
        toJson(shallow(<Button {...defaultOnPress}>Test</Button>))
      ).toMatchSnapshot();
      expect(
        toJson(shallow(<Button {...defaultOnPress} title="qdwdqw" />))
      ).toMatchSnapshot();
      expect(
        toJson(
          shallow(
            <Button
              {...defaultOnPress}
              title="qdwdqw"
              loading={true}
              loadingTitle="My loading title"
            />
          )
        )
      ).toMatchSnapshot();
      expect(
        toJson(
          shallow(<Button {...defaultOnPress} title="qdwdqw" disabled={true} />)
        )
      ).toMatchSnapshot();
    });

    // TODO: uncomment this
    // it('should contain the same title than passed into parameter', () => {
    //   const button1 = shallow(<Button {...defaultOnPress} title="Test" />);
    //   expect(button1.find('Text').text()).toEqual('Test'); // eslint-disable-line

    //   const button2 = shallow(<Button {...defaultOnPress} title="Test2" />);
    //   expect(button2.find('Text').text()).toEqual('Test2'); // eslint-disable-line
    // });

    it('should call onPress when taping the button', () => {
      const onPress = jest.fn();
      const button = shallow(<Button onPress={onPress} />);
      button.find('TouchableOpacity').prop('onPress')();
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onPress).toHaveBeenCalledWith();
    });
  });
});
