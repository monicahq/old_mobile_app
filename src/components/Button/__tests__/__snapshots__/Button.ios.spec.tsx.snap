// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Button IOS should renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.4}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(43, 187, 72)",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "100%",
      },
      false,
    ]
  }
>
  <ForwardRef(Text)
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Test
  </ForwardRef(Text)>
</TouchableOpacity>
`;

exports[`Components Button IOS should renders correctly 2`] = `
<TouchableOpacity
  activeOpacity={0.4}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(43, 187, 72)",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "100%",
      },
      false,
    ]
  }
>
  <ForwardRef(Text)
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    qdwdqw
  </ForwardRef(Text)>
</TouchableOpacity>
`;

exports[`Components Button IOS should renders correctly 3`] = `
<TouchableOpacity
  activeOpacity={0.4}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(43, 187, 72)",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "100%",
      },
      false,
    ]
  }
>
  <ForwardRef(Text)
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    My loading title
  </ForwardRef(Text)>
  <ForwardRef(ActivityIndicator)
    animating={true}
    color="#ffffff"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "marginLeft": 10,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Components Button IOS should renders correctly 4`] = `
<TouchableOpacity
  activeOpacity={0.4}
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(43, 187, 72)",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "100%",
      },
      Object {
        "backgroundColor": "rgba(43, 187, 72, 0.4)",
      },
    ]
  }
>
  <ForwardRef(Text)
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    qdwdqw
  </ForwardRef(Text)>
</TouchableOpacity>
`;
